src/index.js

```diff
@@ -0,0 +1,42 @@
+const BigNumber = require('bignumber.js');
+
+const MoneySafe = Symbol('MoneySafe');
+
+const createCurrency = ({ decimals }) => {
+  const of = (input, value = new BigNumber(String(input))) => {
+    const plus = b => of(value.plus(of(b)));
+    const times = b => of(value.multipliedBy(of(b)));
+    const div = b => of(value.dividedBy(of(b)));
+
+    return Object.assign(plus, {
+      [MoneySafe]: true,
+      constructor: of,
+      map: f => of(f(value)),
```

Comment:
`map` and `valueOf` aren't described in the docs. Maybe describe the `money` object as a monad in the docs, so FP initiates know what's up?